### PR TITLE
Debug mode for children

### PR DESCRIPTION
**Pull request by flops**

### CODE REVIEW

File: packages/browser-proxy/src/index.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ const browserProxyControllerFactory = (transport: ITransport) => {
             pluginName,
             '--config',
             JSON.stringify(config)
-        ]);
+        ], false);
```
Comment (johnthecat) [inline packages/browser-proxy/src/index.ts]: can this argument be optional? looks weird

File: packages/child-process/src/find-ports.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            });
+            server.close();
+        });
+        server.on('error', () => {
+            reject(Error('Can not open any free port on this machine'));
+        });
+    });
+};
+
+export const getAvailablePort = async (ports: Array<number> = [], host: string = '127.0.0.1'): Promise<number> => {
```
Comment (johnthecat) [inline packages/child-process/src/find-ports.ts]: this helper already implemented somewhere in http-server tests or source, can we reuse it?

File: packages/child-process/src/fork.ts [hunk 3]
```diff
@@ -48,10 +62,28 @@ const getExecutor = (filePath: string): string => {
...
-export const fork = (filePath: string, args: Array<string> = []): childProcess.ChildProcess => {
+export async function fork(
+    filePath: string,
+    args: Array<string> = [],
+    debugEnabled: boolean = false
+): Promise<childProcess.ChildProcess> {
+    let processArgs: Array<string> = [];
+
+    if (debugEnabled) {
+        let port = await getAvailablePort(PREFERED_DEBUG_PORTS);
```
Comment (johnthecat) [inline packages/child-process/src/fork.ts]: use const, instead of let

File: packages/child-process/test/find-ports.spec.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    it('should return first preferred available port', async () => {
+        const port = await getAvailablePort([ 45600, 45601 ]);
+
+        chai.expect(port).to.be.equal(45600);
+        chai.expect(await isAvailablePort(port, '127.0.0.1')).to.equal(true);
+    });
+
+    it('should return second preferred available port', async () => {
+        let server = await startServer(45700);
```
Comment (johnthecat) [inline packages/child-process/test/find-ports.spec.ts]: const

File: packages/child-process/test/find-ports.spec.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        server.close();
+    });
+};
+
+describe('find ports', () => {
+    it('should check available port', async () => {
+        const port = 45500;
+        chai.expect(await isAvailablePort(port, '127.0.0.1')).to.equal(true);
+
+        let server = await startServer(port);
```
Comment (johnthecat) [inline packages/child-process/test/find-ports.spec.ts]: const

File: packages/child-process/test/find-ports.spec.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+        chai.expect(port).to.be.equal(45701);
+        chai.expect(await isAvailablePort(port, '127.0.0.1')).to.equal(true);
+
+        await stopServer(server);
+    });
+
+    it('should return random available port', async () => {
+        let server1 = await startServer(45800);
+        let server2 = await startServer(45801);
```
Comment (johnthecat) [inline packages/child-process/test/find-ports.spec.ts]: as above

Comment (DmitryKorlas) [reply packages/child-process/test/find-ports.spec.ts]: guys, const is not a panacea. And may require extra work for switching between `const` -> `let` and vice versa.
https://jamie.build/const

**Pull request merged**